cupy/lib/_routines_poly.py

```diff
@@ -103,6 +103,62 @@ def polymul(a1, a2):
     return cupy.convolve(a1, a2)
 
 
+def polyval(p, x):
+    """Evaluates a polynomial at specific values.
+
+    Args:
+        p (cupy.ndarray or cupy.poly1d): input polynomial.
+        x (scalar, cupy.ndarray): values at which the polynomial
+        is evaluated.
+
+    Returns:
+        cupy.ndarray or cupy.poly1d: polynomial evaluated at x.
+
+    .. warning::
+
+        This function doesn't currently support poly1d objects nor
+        multidimensional ndarrays as values used in evaluation.
+
+    .. seealso:: :func:`numpy.polyval`
+
+    """
+    if isinstance(p, cupy.poly1d):
+        p = p.coeffs
+    if cupy.isscalar(p) or p.ndim == 0:
+        raise TypeError('p can be 1d ndarray or poly1d object only')
+    if p.ndim != 1:
+        # to be consistent with polyarithmetic routines' behavior of
+        # not allowing multidimensional polynomial inputs.
+        raise ValueError('p can be 1d ndarray or poly1d object only')
+        # TODO(Dahlia-Chehata): Support poly1d and multidimensional x
+    if isinstance(x, cupy.poly1d) or (isinstance(x, cupy.ndarray)
+                                      and x.ndim > 1):
+        raise NotImplementedError('poly1d or non 1d values are not'
+                                  ' currently supported')
```

Comment:
Please fix to raise errors for other types.
```suggestion
    if (isinstance(x, cupy.ndarray) and x.ndim <= 1) or numpy.isscalar(x):
        x = cupy.asarray(x)
    else:
        raise NotImplementedError(
            'poly1d or non 1d values are not currently supported')
```